frontend/public/components/overview/build-overview.tsx

```diff
@@ -122,18 +122,27 @@ const BuildOverviewList: React.SFC<BuildOverviewListProps> = ({ buildConfig }) =
   );
 };
 
-export const BuildOverview: React.SFC<BuildConfigsOverviewProps> = ({ buildConfigs }) => (
-  <div className="build-overview">
-    <SidebarSectionHeading text="Builds" />
-    {_.isEmpty(buildConfigs) ? (
-      <span className="text-muted">No Build Configs found for this resource.</span>
-    ) : (
-      _.map(buildConfigs, (buildConfig) => (
-        <BuildOverviewList key={buildConfig.metadata.uid} buildConfig={buildConfig} />
-      ))
-    )}
-  </div>
-);
+export const BuildOverview: React.SFC<BuildConfigsOverviewProps> = ({ buildConfigs, obj }) => {
+  const generatedBy = obj?.metadata?.annotations?.['openshift.io/generated-by'];
+  const isCreatedUsingAddFlows = generatedBy === 'OpenShiftWebConsole';
+
+  if (isCreatedUsingAddFlows && _.isEmpty(buildConfigs)) {
+    return null;
+  }
+
+  return (
+    <div className="build-overview">
+      <SidebarSectionHeading text="Builds" />
+      {_.isEmpty(buildConfigs) ? (
+        <span className="text-muted">No Build Configs found for this resource.</span>
+      ) : (
```

Comment:
can this be removed now?, because if we create an app from `cli`  then it will always have buildconfig created and if we create via `webconsole` and associate pipeline then this component will exit with `null` based on the new condition.

Reply:
> Builds section will not be shown in the sidebar if buildConfig is not present and the app was created using import flows. The latter check is added to make sure that if D/DC was created using CLI/import from yaml then we show the builds section at all times to let the user know that they haven't added a buildConfig.

As I have mentioned in the PR description I have kept this condition since if only a D/DC is created via CLI/Import from yaml we should show the build section so we should not remove this condition. I can do one thing though, since a pipeline can be added to an app using CLI also I can change the above condition `isCreatedUsingAddFlows && _.isEmpty(buildConfigs)` to `pipeline && _.isEmpty(buildConfigs)` so the build section will not be shown only if a pipeline is associated with an app and no buildConfig was found for that app.

Reply:
Yes, It makes sense to conditionally show BC based on the pipeline associated with or not.